src/framework/theme/components/window/window-types.ts

```diff
@@ -0,0 +1,84 @@
+import { TemplateRef, InjectionToken, ViewContainerRef } from '@angular/core';
+import { NbComponentType } from '../cdk/overlay';
+
+export enum NbWindowSize {
+  XXSMALL = 'xxsmall',
+  XSMALL = 'xsmall',
+  SMALL = 'small',
+  MEDIUM = 'medium',
+  LARGE = 'large',
+  XLARGE = 'xlarge',
+  XXLARGE = 'xxlarge',
+}
+
+export enum NbWindowState {
+  MINIMIZED = 'minimized',
+  MAXIMIZED = 'maximized',
+  FULL_SCREEN = 'full-screen',
+}
+
+export interface NbWindowStateChange {
+  oldState: NbWindowState;
+  newState: NbWindowState;
+}
+
+/**
+ * Window configuration options.
+ */
+export class NbWindowConfig {
+  /**
+   * Window title.
+   */
+  title: string = '';
+
+  /**
+   * Initial window state. Full screen by default.
+   */
+  initialState: NbWindowState = NbWindowState.FULL_SCREEN;
+
+  /**
+   * If true than backdrop will be rendered behind window.
+   * By default set to true.
+   */
+  hasBackdrop: boolean = true;
+
+  /**
+   * If set to true mouse clicks on backdrop will close a window.
+   * Default is true.
+   */
+  closeOnBackdropClick: boolean = true;
+
+  /**
+   * If true then escape press will close a window.
+   * Default is true.
+   */
+  closeOnEsc: boolean = true;
+
+  /**
+   * Size of window. Large by default.
+   */
+  size: NbWindowSize = NbWindowSize.LARGE;
```

Comment:
I think the user has to provide two sizes: one for the full-screen state and one for the max. Otherwise, as I understand correctly, in the maximized state window will be full-screen and will occupy the same amount of space but in the right bottom corner.